### ISSUE TITLE
Method to get relative object position (attachments)

### ISSUE THREAD

**Issue by GreenXenith** [minetest/minetest#14696]
### Problem

Attached entities are considered by the server to be at the parent origin. However, the server knows the position and rotation of attachments and parents (aside from armature stuff), so it should be able to solve for child positions/rotation.

The solved relative position should be used for both an object method and raycasting (see #10304).

`object:get_pos()` could natively return this relative position, though that could break some mods. Alternatively a new flag like `object:get_pos(relative)`, or a new method like `object:get_relative_pos()`.

### Solutions

blank

### Alternatives

blank

**Comment by FatalError42O** [comment]
object:get_attach()? that will tell you the position given in :set_attach(). Maybe I misunderstood the issue, but we already can get the relative position of an object. 

It kind of sounds like you want the _global_ position of an attached object? I'm not sure. But either way this is something that can be entirely implemented in lua with existing tools

**Comment by FatalError42O** [comment]
if you mean the position of the object in the global system relative to the object (so attached_pos-parent_pos) then you can just make a rotation matrix (you can use a sight if you need the calculations done) and multiply that by a 3x1 vector, then subtract the object position. https://www.symbolab.com/solver/matrix-multiply-calculator may help

**Comment by appgurueu** [comment]
Yes, this can be implemented in Lua. Since there is no library doing this yet, modders are presently usually just using the incorrect `get_pos` however. A pure Lua implementation also can't really attain the same performance. The engine should provide this, especially since it needs this itself (for raycasts). Related: https://github.com/minetest/minetest/pull/14231 contains code to calculate transformations along the attachment chain.
